lib/logdna.rb

```diff
@@ -82,14 +56,16 @@ def level=(value)
       @level = value
     end
 
-    def log(msg=nil, opts={})
-      loggerExist?
-      message = msg
-      message = yield if msg.nil? && block_given?
-      @response = @@client.buffer(message, default_opts.merge(opts).merge({
+    def log(message, opts={})
+      if(message.length == 0)
+        puts "Your logline cannot be empty"
+        return
+      end
+      message = message.to_s unless message.instance_of? String
+      message = message.encode("UTF-8")
+      @@client.write_to_buffer(message, default_opts.merge(opts).merge({
```

Comment:
So it looks like you no longer check if `@@client` exists (when authorization fails). Is that a problem?

Reply:
> There is no authorization.

I'm not following - previously if `request.basic_auth` failed, `@@client` could be nil. We're not checking for that failure anymore so I'm not sure what happens now.

Reply:
This is the code library that writes a client that connects to our servers. Our servers use the ingestion keys for authenticating. (if a user has it, then he can send data to the server). Basic authentication does not make any sense here.